### PR TITLE
DAOS-7388 doc: Update deployment.md for certs

### PR DESCRIPTION
**Pull request by saurabhtandan**
Update the certificate documentation for 'certs' dir under /etc/daos

Skip-build: true
Skip-test: true

### CODE REVIEW

**Review by dpquigl — DISMISSED** [review]

**Review by dpquigl — DISMISSED** [review]

File: doc/admin/deployment.md [hunk 1]
```diff
@@ -248,7 +248,8 @@ Server nodes require:
 
 After the certificates have been securely distributed, the DAOS configuration files must be
 updated in order to enable authentication and secure communications. These examples assume
-that the configuration and certificate files have been installed under `/etc/daos`:
+that the configuration files have been installed under `/etc/daos` and certificate files under
+`/etc/daos/certs` with root ownership and `755` permission for certs dir:
```
Comment (johannlombardi) [inline doc/admin/deployment.md:252]: Shouldn't this belong to the daos_agent user? Do we really want everyone to be able to read it?

Comment (saurabhtandan) [reply doc/admin/deployment.md:252]: This dir is needed on server side too. It's just that we create it on server side during rpm installation and on client side we don't. So having it somewhere, where both can be targeted was the idea.

Comment (dpquigl) [reply doc/admin/deployment.md:252]: This directory is for both the agent and the server. I might add for the server side the /etc/daos/certs/clients directory as well.

Comment (saurabhtandan) [reply doc/admin/deployment.md:252]: For `/etc/daos/certs/clients` the permissions are 700 right?

Comment (saurabhtandan) [reply doc/admin/deployment.md]: As of now I have mentioned `daos_server` ownership and `700` permissions for `/etc/daos/certs/clients`. If this needs to be changed please let me know.

**Review by dpquigl — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**